src/tomli/_parser.py

```diff
@@ -74,7 +74,12 @@ class TOMLDecodeError(ValueError):
 
 def load(__fp: BinaryIO, *, parse_float: ParseFloat = float) -> dict[str, Any]:
     """Parse TOML from a binary file object."""
-    s = __fp.read().decode()
+    b = __fp.read()
+    if isinstance(b, str):  # type: ignore[unreachable]
+        raise TypeError(
+            "File must be opened in binary mode, e.g. use `open('foo.toml', 'rb')`"
+        )
+    s = b.decode()
```

Comment:
```suggestion
    try:
        s = b.decode()
    except AttributeError:
        raise TypeError(
            "File must be opened in binary mode, e.g. use `open('foo.toml', 'rb')`"
        ) from None
```
Thanks! Could we do this to spare an `isinstance` check in the happy case?